File: 02745.py
N, B = input().split()
B = int(B)

result = 0; ten = 1
for i in range(len(N) - 1, -1, -1):
    if N[i] >= 'A':
        result += ten * (ord(N[i]) - ord('A') + 10)
    else:
        result += ten * (ord(N[i]) - ord('0'))

    ten *= B

print(result)

File: 02745.cpp
#include <iostream>
using namespace std;

int main() {
	string N; int B;
	cin >> N >> B;

	int result = 0, ten = 1;
	for (int i = N.length() - 1; i >= 0; i--) {
		if (N[i] >= 'A')
			result += ten * (N[i] - 'A' + 10);
		else
			result += ten * (N[i] - '0');

		ten *= B;
	}

	cout << result << endl;
}
